### PR TITLE
Bigquery Create View with Options

### LINKED ISSUE #2357: OPTIONS with VIEW not recognized in BQ Dialect
When using `OPTIONS` with `VIEW` it throws an error saying invalid SQL.

This works as expected 
```sql
CREATE OR REPLACE TABLE TEST.test 
OPTIONS( description="test" ) AS 
 select name from TEST2.test
```

linting output `sqlfluff lint ... `
```
L:   1 | P:  30 | L014 | Unquoted identifiers must be consistently upper case.
L:   1 | P:  34 | L001 | Unnecessary trailing whitespace.
L:   2 | P:  21 | L006 | Missing whitespace before =
L:   2 | P:  21 | L006 | Missing whitespace after =
L:   2 | P:  22 | L048 | Missing whitespace before "test"
L:   2 | P:  33 | L001 | Unnecessary trailing whitespace.
L:   3 | P:   2 | L003 | Indentation not consistent with line #2
L:   3 | P:   2 | L010 | Keywords must be upper case.
L:   3 | P:   9 | L014 | Unquoted identifiers must be consistently upper case.
L:   3 | P:  14 | L010 | Keywords must be upper case.
L:   3 | P:  25 | L014 | Unquoted identifiers must be consistently upper case.
L:   4 | P:   1 | L009 | Files must end with a single trailing newline.
```

This doesn't work as expected:

```sql
CREATE OR REPLACE VIEW TEST.test 
OPTIONS( description="test" ) AS 
 select name from TEST2.test
```

Linting output the following:
```                                                                                                        ​
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE OR
                      ​| REPLACE VIEW TEST.test \nOPTION...'
L:   1 | P:  33 | L001 | Unnecessary trailing whitespace.
L:   2 | P:  33 | L001 | Unnecessary trailing whitespace.
L:   4 | P:   1 | L009 | Files must end with a single trailing newline.
```

Interestingly this works

```sql
CREATE OR REPLACE VIEW TEST.test AS
 select name from TEST2.test
```

linting output
```
L:   1 | P:  29 | L014 | Unquoted identifiers must be consistently upper case.
L:   2 | P:   2 | L003 | Indentation not consistent with line #1
L:   2 | P:   2 | L010 | Keywords must be upper case.
L:   2 | P:   9 | L014 | Unquoted identifiers must be consistently upper case.
L:   2 | P:  14 | L010 | Keywords must be upper case.
L:   2 | P:  25 | L014 | Unquoted identifiers must be consistently upper case.
L:   3 | P:   1 | L009 | Files must end with a single trailing newline.
```

## Expected Behaviour

Expected the tool to be able to parse views with options.

## Observed Behaviour
Views with options throw error: 
`Found unparsable section: 'CREATE OR REPLACE VIEW TEST.test \nOPTION...'`

## Steps to Reproduce

Save the example above as file and run `sqlfluff lint `

## Dialect
BigQuery

## Version
sqlfluff, version 0.9.1
Python 3.9.9

## Configuration
```
[sqlfluff]
verbose = 0
nocolor = False
dialect = bigquery
templater = jinja
rules = None
exclude_rules = L042
recurse = 0
output_line_length = 80
runaway_limit = 10
ignore_templated_areas = True
encoding = autodetect
disable_noqa = False
# Comma separated list of file extensions to lint.
# NB: This config will only apply in the root folder.
sql_file_exts = .sql

[sqlfluff:rules]
tab_space_size = 2
max_line_length = 100
indent_unit = space
comma_style = trailing
allow_scalar = True
single_table_references = consistent
unquoted_identifiers_policy = all

# rest is default config
```

### PR DESCRIPTION
**Pull request by tunetheweb**
### Brief summary of the change made

Fixes #2357 

### Are there any other side effects of this change that we should be aware of?
Also cleans up some related ANSI syntax

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_ansi.py [hunk 1]
```diff
@@ -3015,7 +3016,7 @@ class CreateFunctionStatementSegment(BaseSegment):
         Sequence("OR", "REPLACE", optional=True),
         Ref("TemporaryGrammar", optional=True),
         "FUNCTION",
-        Sequence("IF", "NOT", "EXISTS", optional=True),
+        Ref("IfNotExistsGrammar", optional=True),
```
Comment (jpy-git) [inline src/sqlfluff/dialects/dialect_ansi.py:3019]: On line 3016, just above here, it looks like you could also replace
```
Sequence("OR", "REPLACE", optional=True),
```
with
```
Ref("OrReplaceGrammar", optional=True),
```

**Review by jpy-git — APPROVED** [review]
Potential other ANSI line to quickly tidy up, but otherwise looks great 👍

File: src/sqlfluff/dialects/dialect_ansi.py (hunk 1 above)
Comment (tunetheweb) [reply src/sqlfluff/dialects/dialect_ansi.py:3019]: Good spot - done!

Comment (jpy-git) [reply src/sqlfluff/dialects/dialect_ansi.py:3019]: nice, feel free to merge once the ci finishes 👍

**Pull request merged**